base/permuteddimsarray.jl

```diff
@@ -186,4 +186,19 @@ end
     P
 end
 
+function Base.similar(a::PermutedDimsArray{T,N,perm,iperm}, ::Type{S}) where {T,S,N,perm,iperm}
+    new_parent = similar(a.parent, S)
+    PermutedDimsArray{S,N,perm,iperm,typeof(new_parent)}(new_parent)
+end
+
+function Base.similar(a::PermutedDimsArray{T,N,perm,iperm}, ::Type{S}, dims::Dims{N}) where {T,S,N,perm,iperm}
```

Comment:
Do you need to define any more than this one definition? (Do fallbacks not handle the rest?)